Component/LinearSolver/Direct/src/sofa/component/linearsolver/direct/SparseCholeskySolver.inl

```diff
@@ -100,8 +148,75 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
-    N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+
+    sofa::helper::ScopedAdvancedTimer factorization_permTimer("factorization_perm");
+
+    switch (d_type_perm.getValue().getSelectedId() )
+    {
+        case 0:
+        default://None->identity
+            {
+                S = symbolic_Chol (&A) ;		/* ordering and symbolic analysis */
+                N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+            }
+            break;
+
+        case 1://SuiteSparse
+            {
+                int order = -1;
+                S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+                N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+            }
+            break;
+
+        case 2://METIS
+            perm.resize(A.n);
+            iperm.resize(A.n);
+            
+            fill_reducing_perm(A , iperm.data(), perm.data() ); // compute the fill reducing permutation
+    
+            permuted_A = cs_permute( &A , perm.data() , iperm.data() , 1);
+            S = symbolic_Chol( permuted_A ); // symbolic analysis  
```

Comment:
Can it be avoided in case of constant matrix structure?

Reply:
We can, but then it will be locked on that permutation it for all the simulation.

Reply:
This is what we do in `SparseLDLSolver` right? Along with a mechanism to detect some changes in the matrix structure